eoc_journal/eoc_journal.py

```diff
@@ -58,6 +97,66 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_pb_answers(self, all_blocks=False):
+        """
+        Returns a list of dicts with info about all problem builder's pb-answer
+        blocks present in the current coures.
+
+        The items are ordered in the order they appear in the course.
+        """
+        response = self._fetch_pb_answer_blocks(all_blocks)
+        blocks = response['blocks']
+        course_block = blocks[response['root']]
+        result = []
+        for section_id in course_block.get('children', []):
+            section_block = blocks[section_id]
+            for subsection_id in section_block.get('children', []):
+                subsection_block = blocks[subsection_id]
+                for unit_id in subsection_block.get('children', []):
+                    unit_block = blocks[unit_id]
+                    for pb_answer_id in unit_block.get('children', []):
+                        pb_answer_block = blocks[pb_answer_id]
```

Comment:
Tried to find a way to make this less repetitive, but everything I came up with was too cryptic ;)